src/dawn_native/Texture.cpp

```diff
@@ -290,7 +290,7 @@ namespace dawn_native {
 
         // The default value for the view dimension depends on the texture's dimension with a
         // special case for 2DArray being chosen automatically if arrayLayerCount is unspecified.
-        if (desc.dimension == wgpu::TextureViewDimension::Undefined) {
+        //if (desc.dimension == wgpu::TextureViewDimension::Undefined) {
```

Comment:
Please leave this to the original, there must be a reason why only you get this error. Maybe you're upstream with dawn?

Reply:
It is the same on Dawn.

Reply:
This error persists on Linux even with updated node and npm

I was using Node 8.x, which is the canonical Ubuntu version.
I used nvm to switch to Node 12.14.1 (npm 6.13.4)
Same Error:

```
webgpu/index.js:43
              default: throw new Error(msg); break;
                       ^

Error: Validation Error: Invalid value for WGPUTextureViewDimension
    at Immediate.<anonymous> (/home/dcerisano/git/webgpu/index.js:43:30)
    at processImmediate (internal/timers.js:439:21)
```

Reply:
Also who else has built the webgpu NAPI addon for linux?
AFAIK, its only me at this point.
My setup is plain vanilla Ubuntu 18.04.03 LTS

Reply:
Kai found the missing defaults:

https://bugs.chromium.org/p/dawn/issues/detail?id=329#c13

Reply:
If you change the  "texture view dimension" default in dawn.json
to "2D" rather than "undefined" this change is not necessary.